### PR TITLE
Add sysext command

### PR DESCRIPTION
**Pull request by Itxaka**
It will turn the last layer of a given container into a signed sysext ready to use

The idea is, to have a dockerfile and in the last step add the files you want to turn into a sysext, and then call this command to transform them into one

Fixes: https://github.com/kairos-io/kairos/issues/2631
Requires https://github.com/kairos-io/kairos-sdk/pull/362 to be merged before

### CODE REVIEW

**Comment by Itxaka** [comment]
bootable test is failing here but passing locally for some reason ....

File: cmd/sysext.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			cfg.Logger.Logger.Info().Msg("💿 Getting image info")
+
+			platform := fmt.Sprintf("linux/%s", viper.Get("arch"))
+			image, err := utils.GetImage(args[1], platform, nil, nil)
+			if err != nil {
+				cfg.Logger.Logger.Error().Str("image", args[1]).Err(err).Msg("⛔ getting image")
+				return err
+			}
+			// Only for sysext, confext not supported yet
+			AllowList := regexp.MustCompile(`^usr|^/usr`)
```
Comment (mauromorales) [inline cmd/sysext.go:66]: same question here than in the other pr about the dir name

File: cmd/sysext.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+			// Now create the file that tells systemd that this is a sysext!
+			err = os.MkdirAll(filepath.Join(dir, "/usr/lib/extension-release.d/"), os.ModeDir|os.ModePerm)
+			if err != nil {
+				cfg.Logger.Logger.Error().Str("dir", filepath.Join(dir, "/usr/lib/extension-release.d/")).Err(err).Msg("⛔ creating dir")
+			}
+
+			extensionData := "ID=_any\nARCHITECTURE=x86-64"
+
+			if viper.Get("arch") == "arm64" {
+				extensionData = "ID=_any\nARCHITECTURE=arm64"
```
Comment (mauromorales) [inline cmd/sysext.go:83]: how come ARCHITECTURE for amd is x86-64 but for arm is not aarch64?

**Review by mauromorales — APPROVED** [review]

File: cmd/sysext.go (hunk 2 above)
Comment (Itxaka) [reply cmd/sysext.go:83]: Just systemd things 🤣

Comment (Itxaka) [reply cmd/sysext.go:83]: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#ConditionArchitecture=

File: cmd/sysext.go (hunk 1 above)
Comment (Itxaka) [reply cmd/sysext.go]: done!

**Comment by jimmykarily** [comment]
I created an image with this dockerfile:

```
FROM busybox

RUN mkdir -p /usr/lib/extension-release.d/ /usr/local/bin && \
  echo "id=_ANY" > /usr/lib/extension-release.d/extension-release.myfile && \
  echo "myfile content" > /usr/local/bin/myfile
```

```bash
docker build -t sysext-image .`
```

```
./build/enki sysext myfile sysext-image --certificate /home/dimitris/workspace/kairos/enki/keys/db.crt --private-key /home/dimitris/workspace/kairos/enki/keys/db.key
```
and copied that to `/efi/EFI/kairos/active.efi.extra.d/` in a running system (after first doing `mount -o remount,rw /efi`)

I then rebooted and the `/usr/local/bin/myfile` was there.

So it works!
Let me have a quick look at the code too.

File: cmd/sysext.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				cfg.Logger.Logger.Error().Str("dir", filepath.Join(dir, "/usr/lib/extension-release.d/")).Err(err).Msg("⛔ creating dir")
+			}
+
+			arch := "x86-64"
+
+			if viper.Get("arch") == "arm64" {
+				arch = "arm64"
+			}
+
+			extensionData := fmt.Sprintf("ID=_any\nARCHITECTURE=%s", arch)
```
Comment (jimmykarily) [inline cmd/sysext.go:86]: I thought I had to created these manually, nice!

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**